react/features/remote-video-menu/components/web/MuteEveryoneDialog.js

```diff
@@ -0,0 +1,137 @@
+
+declare var APP: Object;
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link MuteEveryoneDialog}.
+ */
+type Props = AbstractProps & {
+
+    /**
+     * The ID of the local participant.
+     */
+    localParticipantId: string,
+
+    /**
+     * The IDs of the remote participants to be muted.
+     */
+    remoteParticipantIds: Array<string>,
+
+     /**
+     * The IDs of the remote participants to exclude from being muted.
+     */
+    exclude: Array<string>,
+};
+
+/**
+ * A React Component with the contents for a dialog that asks for confirmation
+ * from the user before muting a remote participant.
+ *
+ * @extends Component
+ */
+class MuteEveryoneDialog extends AbstractMuteRemoteParticipantDialog<Props> {
+    static defaultProps = {
+        exclude: [],
+        muteLocal: false
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t } = this.props;
+
+        return (
+            <Dialog
+                okKey = 'dialog.muteMultipleParticipantButton'
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.muteMultipleParticipantTitle'
+                width = 'small'>
+                <div>
+                    { t('dialog.muteMultipleParticipantBody') }
+                </div>
+            </Dialog>
+        );
+    }
+
+    _onSubmit: () => boolean;
+
+    /**
+     * Callback to be invoked when the value of this dialog is submitted.
+     *
+     * @returns {boolean}
+     */
+    _onSubmit() {
+        const {
+            dispatch,
+            exclude,
+            localParticipantId,
+            remoteParticipantIds
+        } = this.props;
+
+        remoteParticipantIds
+            .filter(id => !exclude.includes(id))
+            .forEach(id => {
+                if (id === localParticipantId) {
+                    const audioMuted = true;
+
+                    sendAnalytics(createToolbarEvent(AUDIO_MUTE, { enable: audioMuted }));
+                    dispatch(setAudioMuted(audioMuted, true));
+
+                    // FIXME: The old conference logic as well as the shared video feature
+                    // still rely on this event being emitted.
+                    typeof APP === 'undefined'
+                        || APP.UI.emitEvent(UIEvents.AUDIO_MUTED, audioMuted, true);
+                } else {
+                    const props = {
+                        dispatch,
+                        participantID: id
+                    };
+
+                    super._onSubmit.call({ props });
+                }
+            });
+
+        return true;
+    }
+}
+
+/**
+ * Maps part of the redux state to the component's props.
+ *
+ * @param {Object} state - The redux store/state.
+ * @returns {Object}
+ */
+function _mapStateToProps(state: Object) {
+    return {
+        localParticipantId: state['features/base/participants']
```

Comment:
You can use `getLocalParticipant` from base/participants to select the local participant.

Reply:
Didn't know about that - will change, thanks